includes/api/class-give-api.php

```diff
@@ -1037,9 +1037,12 @@ private function get_form_data( $form_info ) {
 			$form['info']['tags'] = get_the_terms( $form_info, 'give_forms_tag' );
 		}
 
-		if ( give_is_setting_enabled( give_get_meta( $form_info->ID, '_give_goal_option', true ) ) ) {
-			$goal_amount = give_get_meta( $form_info->ID, '_give_set_goal', true );
-			$goal_percentage_completed = round( ( give_get_form_earnings_stats( $form_info->ID ) / $goal_amount ) * 100 );
+		// Check whether any goal is to be achieved for the donation form
+		$goal_option = give_get_meta( $form_info->ID, '_give_goal_option', true );
+		$goal_amount = give_get_meta( $form_info->ID, '_give_set_goal', true );
+		if ( give_is_setting_enabled( $goal_option ) && isset( $goal_amount ) &&  $goal_amount > 0 ) {
```

Comment:
@mehul0810 change this to `if ( give_is_setting_enabled( $goal_option ) && $goal_amount  )`